eZ/Publish/Core/REST/Server/Controller/User.php

```diff
@@ -179,16 +179,31 @@ public function loadUser($userId)
         $user = $this->userService->loadUser($userId);
 
         $userContentInfo = $user->getVersionInfo()->getContentInfo();
-        $userMainLocation = $this->locationService->loadLocation($userContentInfo->mainLocationId);
         $contentType = $this->contentTypeService->loadContentType($userContentInfo->contentTypeId);
 
+        $mainLocationId = $userContentInfo->mainLocationId;
+
+        // special case for loading current user, which does not have permissions to load its location.
+        if ($user == $this->repository->getCurrentUser()) {
```

Comment:
@andrerom : I'm not sure I understand.

$user will be the user which is requested for the REST request;
currentUser should be the current logged in user, or anonymous.

With the above, if user with ID 50 tries to load f.e. user with ID 10, the condition will be false?